### PR TITLE
Add tests for change phone number in Boilerplate (#9027)

### LINKED ISSUE #9027: The tests for the phone number changing are missing in the Boilerplate project template
Add tests for change phone number in Boilerplate.Tests

### PR DESCRIPTION
**Pull request by mjebrahimi**
Closes #9027 

## Release Notes

- **New Features**
	- Enhanced user sign-in process with support for both email and phone-based sign-ins.
	- Added functionality for managing phone numbers and email settings within user profiles.
	- Introduced a mock `PhoneService` for testing SMS functionalities.

- **Bug Fixes**
	- Adjusted email confirmation behavior to prevent automatic redirection.

- **Refactor**
	- Improved method naming for clarity in user authentication and management.
	- Streamlined test methods for better maintainability and readability.

- **Tests**
	- Expanded test coverage for email and phone functionalities, including new assertions and helper methods.

### CODE REVIEW

File: src/Templates/Boilerplate/Bit.Boilerplate/src/Tests/PageTests/PageModels/Identity/SettingsPage.Account.Email.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        await Assertions.Expect(emailInput).ToContainTextAsync(userEmail);
+        await Assertions.Expect(Page.GetByPlaceholder(AppStrings.NewEmailPlaceholder)).ToBeVisibleAsync();
+        await Assertions.Expect(Page.GetByRole(AriaRole.Button, new() { Name = AppStrings.Submit })).ToBeVisibleAsync();
+        await Assertions.Expect(Page.GetByRole(AriaRole.Main)).ToContainTextAsync(AppStrings.ConfirmMessageInProfile);
+        await Assertions.Expect(Page.GetByRole(AriaRole.Button, new() { Name = AppStrings.Confirm })).ToBeVisibleAsync();
+    }
+
+    public async Task ChangeEmail(string newEmail)
+    {
+        this.newEmail = newEmail;
```
[the remark this answers is not in the document]
Comment (mjebrahimi) [reply src/Templates/Boilerplate/Bit.Boilerplate/src/Tests/PageTests/PageModels/Identity/SettingsPage.Account.Email.cs:30]: This field is declared in another file for this [partial class](https://github.com/mjebrahimi/bitplatform/blob/d40c44b37409731fdb9fe6b7c31effba59890f05/src/Templates/Boilerplate/Bit.Boilerplate/src/Tests/PageTests/PageModels/Identity/SettingsPage.cs#L9)

File: src/Templates/Boilerplate/Bit.Boilerplate/src/Tests/PageTests/PageModels/Identity/SettingsPage.Account.Email.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    public async Task AssertChangeEmail()
+    {
+        await Assertions.Expect(Page.GetByText(AppStrings.SuccessfulSendChangeEmailTokenMessage)).ToBeVisibleAsync();
+        await Assertions.Expect(Page.GetByRole(AriaRole.Main)).ToContainTextAsync(AppStrings.ConfirmEmailSubtitle);
+        await Assertions.Expect(Page.GetByRole(AriaRole.Main)).ToContainTextAsync(AppStrings.ConfirmEmailMessage);
+        var emailInput = Page.GetByPlaceholder(AppStrings.EmailPlaceholder);
+        await Assertions.Expect(emailInput).ToBeVisibleAsync();
+        await Assertions.Expect(emailInput).ToBeDisabledAsync();
+        await Assertions.Expect(emailInput).ToBeEditableAsync(new() { Editable = false });
```
[the remark this answers is not in the document]
Comment (mjebrahimi) [reply src/Templates/Boilerplate/Bit.Boilerplate/src/Tests/PageTests/PageModels/Identity/SettingsPage.Account.Email.cs:43]: Looks good suggestion 👍

**Review by msynk — APPROVED** [review]

File: src/Templates/Boilerplate/Bit.Boilerplate/src/Tests/PageTests/PageModels/Identity/SignInPage.cs [hunk 3]
```diff
@@ -25,15 +59,42 @@ public override async Task AssertOpen()
...
+        {
+            case Tab.Email:
+                await Page.GetByPlaceholder(AppStrings.EmailPlaceholder).FillAsync(emailOrPhone);
+                break;
+            case Tab.Phone:
+                await Page.GetByPlaceholder(AppStrings.PhoneNumberPlaceholder).FillAsync(emailOrPhone);
+                break;
+            default:
+                throw new ArgumentOutOfRangeException(nameof(tab), tab, null);
+        }
```
[the remark this answers is not in the document]
Comment (mjebrahimi) [reply src/Templates/Boilerplate/Bit.Boilerplate/src/Tests/PageTests/PageModels/Identity/SignInPage.cs:97]: We assert all necessary elements in previously called methods. so here we don't need to assert them again.

File: src/Templates/Boilerplate/Bit.Boilerplate/src/Tests/PageTests/PageModels/Identity/SignInPage.cs [hunk 4]
```diff
@@ -25,15 +59,42 @@ public override async Task AssertOpen()
...
+        return await SignInWithEmail<IdentityHomePage>(email, password);
     }
 
-    public async Task<TPage> SignIn<TPage>(string email = "test@bitplatform.dev", string password = "123456")
+    public async Task<TPage> SignInWithEmail<TPage>(string email = "test@bitplatform.dev", string password = "123456")
         where TPage : IdentityLayout
     {
-        await Page.GetByPlaceholder(AppStrings.EmailPlaceholder).FillAsync(email);
+        return await SignInCore<TPage>(Tab.Email, email, password);
+    }
```
[the remark this answers is not in the document]
Comment (mjebrahimi) [reply src/Templates/Boilerplate/Bit.Boilerplate/src/Tests/PageTests/PageModels/Identity/SignInPage.cs:71]: Nice suggestion good rabbit :) I'm gonna do it.

File: src/Templates/Boilerplate/Bit.Boilerplate/src/Tests/Services/FakePhoneService.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        return message;
+    }
+
+    public static string GetLastOtpFor(string phoneNumber)
+    {
+        var message = GetLastSmsFor(phoneNumber);
+        var otp = Regex.Match(message, @"\d{6}").Value;
+        Assert.IsNotNull(otp, "Otp has not sent");
+        return otp;
+    }
```
[the remark this answers is not in the document]
Comment (mjebrahimi) [reply src/Templates/Boilerplate/Bit.Boilerplate/src/Tests/Services/FakePhoneService.cs:35]: Good one👍

**Review by msynk — APPROVED** [review]

**Pull request merged**